src/Services/RabbitMQService.cs

```diff
@@ -38,6 +39,32 @@ public RabbitMQService(string connectionString, string hostName, string userName
             _model = connectionFactory.CreateConnection().CreateModel();
         }
 
+        public RabbitMQService(string connectionString, string hostName, string exchangeName, string xMatch, string arguments, string userName, string password, int port)
+            : this(connectionString, hostName, userName, password, port)
+        {
+            var result = _model.QueueDeclare();
+            _queueName = result.QueueName;
+
+            var compiledArguments = new Dictionary<string, object>();
+            if (!string.IsNullOrWhiteSpace(xMatch))
+            {
+                compiledArguments.Add("x-match", xMatch);
+            }
+
+            if (!string.IsNullOrWhiteSpace(arguments))
+            {
+                var argumentObject = JObject.Parse(arguments);

```

Comment:
I shall hold my hands up and confess here that I wasn't sure on the best way to pass in the header arguments in order to pass them to QueueBind as a Dictionary. Any advice here would be appreciated!